File: backend-kotlin/src/main/kotlin/com/nbweatherproject/backend/model/WeatherData.kt
package com.nbweatherproject.backend

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "weather_data")
class WeatherData(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "high_temp", nullable = false)
    var highTemp: Double = 0.0,

    @Column(name = "low_temp", nullable = false)
    var lowTemp: Double = 0.0,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var weather: Weather = Weather.UNKNOWN,

    @Column(name = "wind_speed", nullable = false)
    var windSpeed: Double = 0.0,

    @Column(name = "queryId", nullable = false)
    var queryId: Long = 0,

    @Column(nullable = false)
    var date: LocalDate = LocalDate.now()
) {
    enum class Weather {
        CLEAR, MAINLY_CLEAR, PARTLY_CLOUDY, OVERCAST, FOG, DRIZZLE, RAIN, SNOW, RAIN_SHOWERS, SNOW_SHOWERS,
        THUNDERSTORM, UNKNOWN
    }
}

File: backend-ts/src/types/WeatherData.ts
export enum Weather {
  CLEAR = "CLEAR",
  MAINLY_CLEAR = "MAINLY_CLEAR",
  PARTLY_CLOUDY = "PARTLY_CLOUDY",
  OVERCAST = "OVERCAST",
  FOG = "FOG",
  DRIZZLE = "DRIZZLE",
  RAIN = "RAIN",
  SNOW = "SNOW",
  RAIN_SHOWERS = "RAIN_SHOWERS",
  SNOW_SHOWERS = "SNOW_SHOWERS",
  THUNDERSTORM = "THUNDERSTORM",
  UNKNOWN = "UNKNOWN",
}

export interface WeatherData {
  id?: number;
  highTemp: number;
  lowTemp: number;
  weather: string;
  windSpeed: number;
  queryId: number;
  date: Date;
}
